### PR TITLE
[tests-only][full-ci] search: reindex command for spaces

### PR DESCRIPTION
**Pull request by ishabaral**
## Description
Adding test scenario for reindexing spaces via cli

## Related Issue
- Part of https://github.com/owncloud/ocis/issues/9529

## How Has This Been Tested?

- test environment:
- test case 1: locally
- test case 2: CI
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/bootstrap/CliContext.php [hunk 1]
```diff
@@ -118,6 +118,38 @@ public function theAdministratorRemovesAllVersionsOfResources() {
...
 	}
 
+	/**
+	 * @When /^user "([^"]*)" reindexes all spaces using the CLI$/
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 */
+	public function userReindexesAllSpacesUsingTheCli($user) {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/CliContext.php:128]:
```suggestion
	public function userReindexesAllSpacesUsingTheCli($user): void {
```

File: tests/acceptance/features/bootstrap/CliContext.php [hunk 2]
```diff
@@ -118,6 +118,38 @@ public function theAdministratorRemovesAllVersionsOfResources() {
...
+
+	/**
+	 * @When /^user "([^"]*)" reindexes a space with spaceName "([^"]*)" using the CLI$/
+	 *
+	 * @param string $user
+	 * @param string $spaceName
+	 *
+	 * @return void
+	 */
+	public function theAdministratorReindexesASpaceWithSpaceidUsingTheCli(string $user, string $spaceName) {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/CliContext.php:144]:
```suggestion
	public function theAdministratorReindexesASpaceWithSpaceidUsingTheCli(string $user, string $spaceName): void {
```

**Review by saw-jan — COMMENTED** [review]
You could update to other methods where the return type is missing

File: tests/acceptance/features/bootstrap/CliContext.php [hunk 3]
```diff
@@ -139,6 +139,38 @@ public function theAdministratorRemovesTheVersionsOfFileUsingFileId($file, $user
 		$this->featureContext->setResponse(CliHelper::runCommand($body));
 	}
 
+	/**
+	 * @When /^user "([^"]*)" reindexes all spaces using the CLI$/
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/CliContext.php:143]:
```suggestion
	 * @When /^the administrator reindexes all spaces using the CLI$/
```

File: tests/acceptance/features/bootstrap/CliContext.php [hunk 4]
```diff
@@ -139,6 +139,38 @@ public function theAdministratorRemovesTheVersionsOfFileUsingFileId($file, $user
...
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 */
+	public function userReindexesAllSpacesUsingTheCli($user): void {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/CliContext.php:149]:
```suggestion
	 *
	 * @return void
	 */
	public function userReindexesAllSpacesUsingTheCli(): void {
```

File: tests/acceptance/features/bootstrap/CliContext.php [hunk 5]
```diff
@@ -139,6 +139,38 @@ public function theAdministratorRemovesTheVersionsOfFileUsingFileId($file, $user
...
+		$command = "search index --all-spaces";
+		$body = [
+			"command" => $command
+		];
+		$this->featureContext->setResponse(CliHelper::runCommand($body));
+	}
+
+	/**
+	 * @When /^user "([^"]*)" reindexes a space with spaceName "([^"]*)" using the CLI$/
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/CliContext.php:158]:
```suggestion
	 * @When /^the administrator reindexes a space "([^"]*)" using the CLI$/
```

File: tests/acceptance/features/bootstrap/CliContext.php [hunk 6]
```diff
@@ -139,6 +139,38 @@ public function theAdministratorRemovesTheVersionsOfFileUsingFileId($file, $user
...
+	 *
+	 * @param string $user
+	 * @param string $spaceName
+	 *
+	 * @return void
+	 */
+	public function userReindexesASpaceWithSpacenameUsingTheCli(string $user, string $spaceName): void {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/CliContext.php:165]:
```suggestion
	 * @param string $spaceName
	 *
	 * @return void
	 */
	public function userReindexesASpaceWithSpacenameUsingTheCli(string $spaceName): void {
```

File: tests/acceptance/features/bootstrap/CliContext.php [hunk 7]
```diff
@@ -139,6 +139,38 @@ public function theAdministratorRemovesTheVersionsOfFileUsingFileId($file, $user
...
+		$spaceId = $this->featureContext->spacesContext->getSpaceIdByName($user, $spaceName);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/CliContext.php:166]:
```suggestion
		$spaceId = $this->featureContext->spacesContext->getSpaceIdByName(<admin-user>, $spaceName);
```

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/cliCommands/searchReIndex.feature [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+
+  Scenario: reindex all spaces
+    When user "Alice" reindexes all spaces using the CLI
+    Then the command should be successful
+
+
+  Scenario: reindex a space
+    When user "Alice" reindexes a space with spaceName "new-space" using the CLI
+    Then the command should be successful
```
Comment (ScharfViktor) [inline tests/acceptance/features/cliCommands/searchReIndex.feature:21]: I would check that search works after reindex
maybe we can add step where we found `textfile.txt` file?

File: tests/acceptance/features/bootstrap/CliContext.php [hunk 9]
```diff
@@ -139,6 +139,36 @@ public function theAdministratorRemovesTheVersionsOfFileUsingFileId($file, $user
...
+
+	/**
+	 * @When /^the administrator reindexes a space "([^"]*)" using the CLI$/
+	 *
+	 * @param string $spaceName
+	 *
+	 * @return void
+	 */
+	public function theAdministratorReindexesASpaceWithSpacenameUsingTheCli(string $spaceName): void {
+		$spaceId = $this->featureContext->spacesContext->getSpaceIdByName("admin", $spaceName);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/CliContext.php:164]:
```suggestion
		$spaceId = $this->featureContext->spacesContext->getSpaceIdByName($this->featureContext->getAdminUsername(), $spaceName);
```

File: tests/acceptance/features/cliCommands/searchReIndex.feature [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    When the administrator reindexes all spaces using the CLI
+    Then the command should be successful
+    When user "Alice" searches for "textfile.txt" using the WebDAV API
+    Then the HTTP status code should be "207"
+    And the search result of user "Alice" should contain only these entries:
+      | /textfile.txt |
+
+
+  Scenario: reindex a space
+    Given user "Alice" has created the following tags for file "textfile.txt" of the space "new-space":
```
Comment (saw-jan) [inline tests/acceptance/features/cliCommands/searchReIndex.feature:26]:
```suggestion
  Scenario: reindex a space
    Given using spaces DAV path
    And user "Alice" has created the following tags for file "textfile.txt" of the space "new-space":
```

File: tests/acceptance/features/bootstrap/CliContext.php [hunk 11]
```diff
@@ -139,6 +139,35 @@ public function theAdministratorRemovesTheVersionsOfFileUsingFileId($file, $user
...
+
+	/**
+	 * @When /^the administrator reindexes a space "([^"]*)" using the CLI$/
+	 *
+	 * @param string $spaceName
+	 *
+	 * @return void
+	 */
+	public function theAdministratorReindexesASpaceWithSpacenameUsingTheCli(string $spaceName): void {
+		$spaceId = $this->featureContext->spacesContext->getSpaceIdByName($this->featureContext->getAdminUsername(), $spaceName);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/CliContext.php:163]: `spacesContext` is available in this context
```suggestion
		$spaceId = $this->spacesContext->getSpaceIdByName($this->featureContext->getAdminUsername(), $spaceName);
```

File: tests/acceptance/features/bootstrap/CliContext.php [hunk 12]
```diff
@@ -139,6 +139,35 @@ public function theAdministratorRemovesTheVersionsOfFileUsingFileId($file, $user
...
+	}
+
+	/**
+	 * @When /^the administrator reindexes a space "([^"]*)" using the CLI$/
+	 *
+	 * @param string $spaceName
+	 *
+	 * @return void
+	 */
+	public function theAdministratorReindexesASpaceWithSpacenameUsingTheCli(string $spaceName): void {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/CliContext.php:162]:
```suggestion
	public function theAdministratorReindexesASpaceUsingTheCli(string $spaceName): void {
```

**Review by saw-jan — APPROVED** [review]
:+1:

**Pull request merged**